app/src/main/java/org/fossasia/susi/ai/helper/DateTimeHelper.kt

```diff
@@ -85,6 +90,25 @@ object DateTimeHelper {
         val sdf = SimpleDateFormat("hh:mm aaa")
         val tz = TimeZone.getDefault()
         sdf.timeZone = tz
-        return sdf.format(formatDate(date))
+        return if (date.isEmpty()) {
+            ""
+        } else {
+            sdf.format(formatDate(date))
+        }
+    }
+
+    fun formatDate(timestamp: String?, months: Array<String>): String? {
+        return if (timestamp != null && timestamp.length > 10 && months.size == 12) {
+            var date: String? = ""
+            timestamp.trim()
+            val month = timestamp.substring(5, 7).toInt()
+            date = timestamp.substring(8, 10) + " " +
+                    months[month - 1].toString() +
+                    ", " + timestamp.substring(0, 4)
+
+            date
+        } else {
+            ""
```

Comment:
Remove this

Reply:
What exactly?? timestamp!=null can be removed by making timestamp non nullable, if else condition can't be removed because if not checked it'll expose indexOutOfBound exceptions in months and string length(So if this happens just return empty string ie don't print anything in date).

Reply:
Throw error instead

Reply:
Thrown error and added corresponding tests, There wont be these redundancies from next time.
please notice updated screenshot also.